### ISSUE TITLE
[Azure] Can we re-enable deployments into existing Azure resource groups?

### ISSUE THREAD

**Issue by rfranzke** [gardener/gardener-extension-provider-azure#5]
Some time back we have disabled deployments into [existing Azure resource groups and existing Azure VNets](https://github.com/gardener/gardener/blob/72d4693ab5da13f9512e963bd5e8aa5772738035/pkg/apis/garden/validation/validation.go#L696-L700) because the Azure cloud provider implementation did not clean up self-created resources properly (tested with version 1.7.6).

Has the Azure cloud provider been improved in that regards so that we can re-enable it again?

Status:
 - [x] Enable deploying shoots into existing vNETs.
ref: https://github.com/gardener/gardener/pull/1558 https://github.com/gardener/gardener-extensions/pull/371
 - [ ] Enable deploying shoots into existing resourceGroups

**Comment by muenchdo** [comment]
We would be interested in this as we are currently in discussions with a customer about deploying a shoot into their Azure subscription. Being able to create resource groups seems to require quite broad permissions which we would like to avoid.

**Comment by dkistner** [comment]
As far a I recall correctly the only argument against Shoot deployment into existing resource groups are the left overs which could remain on the infrastructure when the cluster is deleted. Unfortunately the Azure cloud provider have still issues with that.
So yes, I think we can discuss to enable that. 
Btw: PRs for that are warmly welcome :)

**Comment by muenchdo** [comment]
So you would say it could be enabled again and there would be a disclaimer somewhere that it can result in orphaned resources?

**Comment by dkistner** [comment]
Yes, something like this I could image. Users have to be aware that some left overs can remain if the cluster is gone.

**Comment by rfranzke** [comment]
@muenchdo wanna give it a try? If you need help just ping, happy to guide you
